src/tests/ftest/cart/util/daos_sys_logscan.py

```diff
@@ -0,0 +1,553 @@
+            term = self._cur_term[puuid]
+            self._pools[puuid][term]["maps"][to_ver] = {
+                "carryover": False,
+                "from_ver": from_ver,
+                "time": datetime,
+                "rb_gens": {}
+            }
+            if self._debug:
+                print(f"FOUND pool {puuid} map update {from_ver}->{to_ver} rank {rank}\t{host}\t"
+                      f"PID {pid}\t{fname}")
+            return True
+        return False
+
+    def _get_rb_components(self, match):
+        return match.group(1), int(match.group(2)), int(match.group(3)), match.group(4)
+
+    def _match_ps_rb_start(self, fname, line, pid, rank):
+        # Do not match on new rebuild log format if we found legacy format
+        if not self._check_rb_new_fmt:
+            return False
+        msg, host, datetime = self._get_line_components(line)
+        match = self.re_rebuild_ldr_start.match(msg)
+        if match:
+            # Disable checking for legacy rebuild log format, to save execution time
+            self._check_rb_legacy_fmt = False
+            puuid, ver, gen, op = self._get_rb_components(match)
+            if not self._is_leader(puuid, rank, pid):
+                return True
+            term = self._cur_term[puuid]
+            if term < 1:
+                self._warn(f"pool {puuid} I don't know what term it is ({term})!", fname, line)
+                return True
+            if gen in self._pools[puuid][term]["maps"][ver]["rb_gens"]:
+                self._warn(f"pool {puuid} term {term} ver {ver} already has gen {gen}", fname, line)
+            # Future possibility: keep timestamps, durations for scan start, pull start, completed
+            self._pools[puuid][term]["maps"][ver]["rb_gens"][gen] = {
+                "op": op,
+                "start_time": datetime,
+                "time": "xx/xx-xx:xx:xx.xx",
+                "started": True,
+                "scanning": False,
+                "pulling": False,
+                "completed": False,
+                "aborted": False,
+                "duration": 0
+            }
+            if self._debug:
+                print(f"{datetime} FOUND rebuild start in term {term}, rb={puuid}/{ver}/{gen}/{op} "
+                      f"rank {rank}\t{host}\tPID {pid}\t{fname}")
+            return True
+        return False
+
+    def _match_legacy_ps_rb_start(self, fname, line, pid, rank):
+        # Do not match on legacy rebuild log format if we found new format
+        if not self._check_rb_legacy_fmt:
+            return False
+        msg, host, datetime = self._get_line_components(line)
+        match = self.re_old_ldr_start.match(msg)
+        if match:
+            # Disable checking for new rebuild log format, to save execution time
+            self._check_rb_new_fmt = False
+            puuid = match.group(1)
+            ver = int(match.group(2))
+            gen = int(match.group(3))
+            op = match.group(4)
```

Comment:
Nit, could be replaced with:
`puuid, ver, gen, op = self._get_rb_components(match)`